honeybadger-io/honeybadger-js
# feat(rollup-plugin): add rollup/vite plugin that uploads sourcemaps to honeybadger API

## Status
WIP

## Description
This PR introduces a new rollup/vite plugin to upload sourcemaps to Honeybadger. Addresses https://github.com/honeybadger-io/honeybadger-js/issues/992. I have tested that it uploads sourcemaps and that those sourcemaps are applied correctly in the Honeybadger UI. 

It is not yet feature-complete as compared to the webpack plugin. The main functionality missing is sending deploy notifications. The webpack plugin also has configurable parallelism, which this does not currently have.

We also may want to add a check to make sure we're only uploading sourcemaps on production builds. 

## Todos

## Steps to Test or Reproduce
There are two example projects added here. One uses rollup to package some simple JS, and the other is a Vite/Vue setup. To test:
* run `npm run build` in the plugin folder to build the plugin
* run `npm run build` in the example folder to build the example project
If you haven't configured your API key correctly, you should get a nice, useful error. 
Configure your env variables using `.env.local`. Each example project includes a `.env.example` you can copy from.
